neard/src/config.rs

```diff
@@ -426,6 +430,8 @@ pub struct Config {
     pub epoch_sync_enabled: bool,
     #[serde(default = "default_view_client_throttle_period")]
     pub view_client_throttle_period: Duration,
+    #[serde(default = "default_trie_viewer_state_size_limit")]
+    pub trie_viewer_state_size_limit: Option<u64>,
 }
```

Comment:
Unrelated to the PR, but the odd thing I've noticed is that we use both `#[serde(default)]` on the struct itself and `#[serde(default = "default_trie_viewer_state_size_limit")]`  on the field. That's tautological and can lead to an inconsistency between `Default` impl and the defaults during deserialization. I think for this particular struct we want to remove all `sered(default` from fields and just inline the the default functions into `impl Default for Config`. 

In a similar vein, some structs in this file (eg Network) have `impl Default` but do not have struct-level `serde(default)`. This also feels odd a bit.

Reply:
Yeah, clearly needs some look. Should not be part of this PR, though, but good call!